dependencies.yaml

```diff
@@ -276,12 +293,23 @@ dependencies:
           # Align nvcomp version with rapids-cmake
           - nvcomp==3.0.6
           - spdlog>=1.12.0,<1.13
-  build_wheels:
+  rapids_build_skbuild:
     common:
+      - output_types: conda
+        packages:
+          - rapids-build-backend >=0.3.0,<0.4.0.dev0
```

Comment:
1. Can we move rapids-build-backend to a separate list for `output_types: [conda, requirements, pyproject]`? It is currently being repeated in two separate lists (conda; requirements/pyproject).
2. Can we define this version/value with an anchor and reuse the value in `rapids_build_setuptools`?

Reply:
Yes and yes! Great points.

I'd had them separate when I was originally using slightly different forms (`~={floor}` for conda, `>={floor},<{ceiling}` for pyproject/requirements).

I'll consolidate them.

Reply:
This is done in recent commits.

While doing it, I also saw some other areas where the approach could be simplified. I'd been working under the (assumed) constraint that we would prefer this pattern for conda environments:

```text
rmm=24.8.*
```

to this

```text
rmm==24.8.*,>=0.0.0a0
```

And so that led to me introducing some similar splits in `dependencies.yaml` to the one this comment thread is about, **just for the purpose of expressing the constraint differently in conda vs. non-conda contexts**.

But I guess based on the changes I see in https://github.com/rapidsai/raft/pull/2331 that we're find with that alpha spec being added even for conda requirements. So I've consolidated those blocks back together.

If you look at the diff for `dependencies.yaml` on this PR now, it's much smaller and simpler.